net/Socket.cpp

```diff
@@ -1239,50 +1239,58 @@ std::shared_ptr<Socket> ServerSocket::accept()
 #else
     const int rc = fakeSocketAccept4(getFD());
 #endif
-    LOG_TRC("Accepted socket #" << rc << ", creating socket object.");
-    try
+    if (rc < 0)
     {
-        // Create a socket object using the factory.
-        if (rc != -1)
+        const int cause = errno;
+        if( cause != EINTR && cause != EAGAIN && cause != EWOULDBLOCK )
         {
-#if !MOBILEAPP
-            char addrstr[INET6_ADDRSTRLEN];
+            // Unrecoverable error, throw, which will close `ServerSocket` and discard it from `AcceptPoll` during `SocketPoll::poll`
+            const std::string err_msg = ErrorMessagePrefix + std::to_string(cause) + ", " + std::strerror(cause) + ')';
+            LOG_ERR(err_msg);
+            throw std::runtime_error(err_msg);
+        }
+        return nullptr;
+    }
+    LOG_TRC("Accepted socket #" << rc << ", creating socket object.");
 
-            Socket::Type type;
-            const void *inAddr;
-            if (clientInfo.sin6_family == AF_INET)
-            {
-                struct sockaddr_in *ipv4 = (struct sockaddr_in *)&clientInfo;
-                inAddr = &(ipv4->sin_addr);
-                type = Socket::Type::IPv4;
-            }
-            else
-            {
-                struct sockaddr_in6 *ipv6 = &clientInfo;
-                inAddr = &(ipv6->sin6_addr);
-                type = Socket::Type::IPv6;
-            }
+#if !MOBILEAPP
+    char addrstr[INET6_ADDRSTRLEN];
 
-            std::shared_ptr<Socket> _socket = createSocketFromAccept(rc, type);
+    Socket::Type type;
+    const void *inAddr;
+    if (clientInfo.sin6_family == AF_INET)
+    {
+        struct sockaddr_in *ipv4 = (struct sockaddr_in *)&clientInfo;
+        inAddr = &(ipv4->sin_addr);
+        type = Socket::Type::IPv4;
+    }
+    else
+    {
+        struct sockaddr_in6 *ipv6 = &clientInfo;
+        inAddr = &(ipv6->sin6_addr);
+        type = Socket::Type::IPv6;
+    }
+    ::inet_ntop(clientInfo.sin6_family, inAddr, addrstr, sizeof(addrstr));
+#endif
```

Comment:
Lets have the try{ } inside the non-mobile conditionals - rather than adding more conditionals. The FakeSocket is unlikely to have these problems - and the change is not necessary.

Reply:
OK